### PR TITLE
1833

### LINKED ISSUE #1586: Leetcode Question Number 1833
Leetcode Question Number 1833 solution needs to be added in the repo . 
 The link for the question is [this](https://leetcode.com/problems/maximum-ice-cream-bars).
 Accepted Languages C++ / JAVA . 
 You Only need to add the  class Solution in the file. 
 Make Sure you have read the contributions guidelines before creating an PR.

### PR DESCRIPTION
**Pull request by classerase**
Fixes #1586 

My code sample is in : 
- [x] CPP
- [ ] Java 

Screenshot of working code sample  : 
image

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My code sample is formatted and works correctly 

[Contributor Guide]: https://github.com/amanv8060/Leetcode-Questions/blob/main/CONTRIBUTING.md

PS: Sorry about all of these commits, the ones to do with 4.cpp are from my main branch, and since the code wont change it doesnt affect it. the other commits apart from 1833 are me trying to get the latest changes from your branch so i can open a pull request safely.

### CODE REVIEW

File: solutions/4.cpp [hunk 1]
```diff
@@ -1,3 +1,20 @@
+<<<<<<< HEAD
```
Comment (amanv8060) [inline solutions/4.cpp:1]: looks like merge conflict not handled properly

**Review by amanv8060 — CHANGES_REQUESTED** [review]
Please delete 4.cpp

File: solutions/4.cpp (hunk 1 above)
Comment (classerase) [reply solutions/4.cpp:1]: yeah. ill delete those lines really quickly

**Comment by classerase** [comment]
better now?

**Review by amanv8060 — APPROVED** [review]

**Comment by amanv8060** [comment]
@classerase
> better now?
No , looks like this is a wrong answer

File: solutions/1833.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    int able_to_buy = 0;
+
+    for (int i = 0; i <= costs.size(); i++) {
+      if (coins - costs[i] >= 0) {
+        able_to_buy++;
+        coins -= costs[i];
+      }
+    }
+
+    return --able_to_buy;
```
Comment (amanv8060) [inline solutions/1833.cpp:18]: remove --

File: solutions/1833.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+using namespace std;
+
+class Solution {
+ public:
+  int maxIceCream(vector<int>& costs, int coins) {
+    sort(costs.begin(), costs.end());
+
+    int able_to_buy = 0;
+
+    for (int i = 0; i <= costs.size(); i++) {
```
Comment (amanv8060) [inline solutions/1833.cpp:11]: i < costs.size()

**Review by amanv8060 — CHANGES_REQUESTED** [review]
Looks like some more changes are required

File: solutions/1833.cpp (hunk 3 above)
Comment (classerase) [reply solutions/1833.cpp:11]: its working on the test cases the website gives

Comment (classerase) [reply solutions/1833.cpp:11]: but yeah, np ill make the changes

**Comment by classerase** [comment]
good now?

File: solutions/1833.cpp (hunk 3 above)
Comment (amanv8060) [reply solutions/1833.cpp]: in a general case ,you cant acc vector[vector.size()]

**Review by amanv8060 — APPROVED** [review]

**Comment by amanv8060** [comment]
thanks for the contribution.

**Comment by classerase** [comment]
no problem :)

**Pull request merged**